### PR TITLE
Fix issue 6798

### LINKED ISSUE #6798: d2g: chain of trust should include original task definition in chain-of-trust.json
When a docker worker chain of trust task is executed by generic-worker, the task payload that appears in the chain-of-trust.json file should be the original docker worker task payload, not the converted version.

### PR DESCRIPTION
**Pull request by petemoore**
Fixes #6798

First commit is a test, which should fail, to demonstrate the issue.
Second commit is a fix, which should hopefully pass the CI!

### CODE REVIEW

File: workers/generic-worker/d2g_multiuser_test.go [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+		Image:   json.RawMessage(`"denolehov/curl"`),
+		Features: dockerworker.FeatureFlags{
+			ChainOfTrust: true,
+		},
+		MaxRunTime: 10,
+	}
+	defaults.SetDefaults(&payload)
+	td := testTask(t)
+
+	switch fmt.Sprintf("%s:%v", runtime.GOOS, config.RunTasksAsCurrentUser) {
```
Comment (matt-boris) [inline workers/generic-worker/d2g_multiuser_test.go:29]: I like this switch statement! Nice

File: workers/generic-worker/d2g_multiuser_test.go [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+		if !reflect.DeepEqual(*cotCertTaskPayload, payload) {
+			t.Fatalf("Expected chain of trust cert task payload to match original docker worker payload...\nOriginal:\n%#v\nChain of trust version:\n%#v", payload, cotCertTaskPayload)
+		}
+		if !reflect.DeepEqual(cotCert.Task.Scopes, td.Scopes) {
+			t.Fatalf("Expected chain of trust cert task scopes to match original docker worker scopes...\nOriginal:\n%#v\nChain of trust version:\n%#v", cotCert.Task.Scopes, td.Scopes)
+		}
+	default:
+		_ = submitAndAssert(t, td, payload, "exception", "malformed-payload")
+	}
+	t.Log(LogText(t))
```
Comment (matt-boris) [inline workers/generic-worker/d2g_multiuser_test.go:53]: Should we log out the logtext each time the test is run?

**Review by matt-boris — COMMENTED** [review]
LGTM Pete, thanks. Left one small question about the test.

File: workers/generic-worker/d2g_multiuser_test.go (hunk 2 above)
Comment (petemoore) [reply workers/generic-worker/d2g_multiuser_test.go:53]: See https://pkg.go.dev/testing#T.Log - logs only printed if test fails ;-)

Comment (matt-boris) [reply workers/generic-worker/d2g_multiuser_test.go:53]: Ah perfect, my bad!

**Review by matt-boris — APPROVED** [review]

**Pull request merged**